src/main/java/seedu/duke/command/AddIncomeCommand.java

```diff
@@ -54,6 +55,8 @@ private Goal handleGoal() throws DukeException {
         String goal = getArg(GOAL_ARG);
         if (goal == null) {
             return state.getUncategorisedGoal();
+        } else if (goal.equalsIgnoreCase(UNCATEGORISED)) {
```

Comment:
Add same check for AddExpenseCommand, otherwise "uncategorised" category will be created.